mlx_audio/tts/models/llama/llama.py

```diff
@@ -294,31 +317,59 @@ def generate(
         split_pattern: str = "\n",
         max_tokens: int = 1200,
         verbose: bool = False,
+        ref_audio: mx.array = None,
+        ref_text: Optional[str] = None,
         **kwargs,
     ):
         prompt = text.replace("\\n", "\n").replace("\\t", "\t")
         prompts = prompt.split(split_pattern)
-        prompts = [f"{voice}: " + p for p in prompts]
 
-        all_input_ids = []
-
-        for prompt in prompts:
-            input_ids = mx.array(self.tokenizer(prompt, return_tensors="pt").input_ids)
-            all_input_ids.append(input_ids)
+        audio_input_ids = None
+        if ref_audio is not None and ref_text is not None:
+            audio_input_ids = encode_audio_to_codes(ref_audio) + 128266
+            audio_transcript_ids = mx.array(
+                self.tokenizer(ref_text, return_tensors="pt").input_ids
```

Comment:
```suggestion
                self.tokenizer(ref_text, return_tensors="mlx").input_ids
```

Reply:
Here we should have either `mlx` or `numpy` as return_tensors whenever possible so that when we remove `torch` as a dependency out models don't break for users that don't have it installed.